webapp/channels/src/components/admin_console/billing/billing_subscriptions/to_paid_plan_nudge_banner.tsx

```diff
@@ -0,0 +1,242 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import {useIntl, FormattedMessage} from 'react-intl';
+import moment from 'moment';
+
+import AlertBanner from 'components/alert_banner';
+import useOpenPricingModal from 'components/common/hooks/useOpenPricingModal';
+import useOpenCloudPurchaseModal from 'components/common/hooks/useOpenCloudPurchaseModal';
+import useOpenSalesLink from 'components/common/hooks/useOpenSalesLink';
+import AnnouncementBar from 'components/announcement_bar/default_announcement_bar';
+
+import {getSubscriptionProduct as selectSubscriptionProduct} from 'mattermost-redux/selectors/entities/cloud';
+import {getCurrentUser, isCurrentUserSystemAdmin} from 'mattermost-redux/selectors/entities/users';
+import {savePreferences} from 'mattermost-redux/actions/preferences';
+import {cloudFreeDeprecationUIEnabled, get as getPreference} from 'mattermost-redux/selectors/entities/preferences';
+
+import {AnnouncementBarTypes, CloudBanners, CloudProducts, Preferences} from 'utils/constants';
+import {t} from 'utils/i18n';
+
+import {GlobalState} from '@mattermost/types/store';
+
+import './to_paid_plan_nudge_banner.scss';
+
+enum DismissShowRange {
+    GreaterThanEqual90 = '>=90',
+    BetweenNinetyAnd60 = '89-61',
+    SixetyTo31 = '60-31',
+    ThirtyTo11 = '30-11',
+    TenTo1 = '10-1',
+    Zero = '0'
+}
+
+const cloudFreeCloseMoment = '20230327'; // TBD, final day of cloud free
+
+interface ToPaidPlanDismissPreference {
+
+    // range represents the range for the days to the deprecation of cloud free e.g. in 30 to 10 days to deprecate cloud free
+    // Incase of dismissing the banner, range represents the time (days) period when this banner was dismissed.
+    // This is important because in case the banner was dismissed for a certain period, it helps us know that we should not show it again for that period.
+    range: DismissShowRange;
+    show: boolean;
+}
+
+export const ToPaidPlanBannerDismissable = () => {
+    const dispatch = useDispatch();
+
+    const openPricingModal = useOpenPricingModal();
+
+    const currentUser = useSelector(getCurrentUser);
+    const isAdmin = useSelector(isCurrentUserSystemAdmin);
+    const product = useSelector(selectSubscriptionProduct);
+    const enabled = useSelector(cloudFreeDeprecationUIEnabled);
+    const currentProductStarter = product?.sku === CloudProducts.STARTER;
+
+    const now = moment(Date.now());
```

Comment:
Nit:

Very minor and likely not very noticeable, but we could instead do:

```typescript
const now = moment().startOf('day')
```

Since the current implementation will include the time (rather than just the date), the time will be a part of the diff. 

So, if the date string that we have set here for cloud to be deprecated was `2023-03-30` with today being `2023-03-29`, the `daysToCloudFreeEnd` would actually appear to be 0, even if there is some time left in the day until deprecation.

It might cause some confusion when the banner message states `Cloud Starter will be deprecated in 0 days. Upgrade to a paid plan or contact sales.`

This might also require a change here: https://github.com/mattermost/mattermost-server/pull/22622/files#diff-cf8060a56e72d942b4fe0feb8925d56ebe7430e89817c2510c4aa0d4aa3222b5R154

To check if `daysToCloudFreeEnd == 0` rather than `daysToCloudFreeEnd < 0` so we display a message stating that the plan was deprecated once we hit 0 days.